File: backend/app/models/project.py
from sqlalchemy import Column, Integer, Float, String, DateTime
from datetime import datetime
from app.db.base import Base


class Project(Base):
    __tablename__ = "projects"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, unique=True, index=True)
    developerEntity = Column(String)
    ecosystemType = Column(String)
    location = Column(String)
    status = Column(String, default="pending")
    totalCredits = Column(Float, default=0.0)
    createdAt = Column(DateTime, default=datetime.utcnow)
    tx_hash = Column(String, nullable=True)
    ipfs_cid = Column(String, nullable=True)

File: frontend/src/types/project.ts
export interface Project {
  id: number;
  name: string;
  developer?: string;
  developerEntity?: string;
  location: string;
  ecosystemType: 'Rhizophora' | 'Avicennia' | 'Sonneratia' | 'Mixed Species';
  status: 'draft' | 'pending' | 'verified' | 'retired';
  totalCredits: number;
  createdAt: string;
  tx_hash?: string;
  ipfs_cid?: string;
}

export interface ProjectListResponse {
  projects: Project[];
  total: number;
  page: number;
  pageSize: number;
}